### PR TITLE
[unit test] Add unit test for FlowCounterRouteOrch

### PR DESCRIPTION
**Pull request by keboliu**
Signed-off-by: kebol@nvidia.com

**What I did**
Add unit test framework for FlowCounterRouteOrch
Add unit test case to cover the route pattern add/delete flow

**Why I did it**
The unit test framework is not available for FlowCounterRouteOrch which will block the developers to add UT to cover the code change in the future. So add the unit test framework for it. Add also add a test to cover the route pattern add/delete flow. 

**How I verified it**
run unit test and all of the new test can pass.

**Details if related**

### CODE REVIEW

**Review by Junchao-Mellanox — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@keboliu  well done on adding more tests to swss.

File: tests/mock_tests/flowcounterrouteorch_ut.cpp [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+            ASSERT_EQ(gNeighOrch, nullptr);
+            gNeighOrch = new NeighOrch(m_app_db.get(), APP_NEIGH_TABLE_NAME, gIntfsOrch, gFdbOrch, gPortsOrch, m_chassis_app_db.get());
+
+            auto* tunnel_decap_orch = new TunnelDecapOrch(m_app_db.get(), APP_TUNNEL_DECAP_TABLE_NAME);
+            vector<string> mux_tables = {
+                CFG_MUX_CABLE_TABLE_NAME,
+                CFG_PEER_SWITCH_TABLE_NAME
+            };
+            auto* mux_orch = new MuxOrch(m_config_db.get(), mux_tables, tunnel_decap_orch, gNeighOrch, gFdbOrch);
```
Comment (prsunny) [inline tests/mock_tests/flowcounterrouteorch_ut.cpp:183]: In unit test, do we need to initialize those orch that are not being used? Suggest remove all other orchs that are not really required.

Comment (keboliu) [reply tests/mock_tests/flowcounterrouteorch_ut.cpp:183]: @prsunny actually NeighOrch has a dependency on MuxOrch, see below code. W/O MuxOrch we will face segment fault when running the test. 
https://github.com/Azure/sonic-swss/blob/18c73a196704d7feb740fc13b15242a8aa30c5dc/orchagent/neighorch.cpp#L809
```
bool NeighOrch::addNeighbor(const NeighborEntry &neighborEntry, const MacAddress &macAddress)
{
    SWSS_LOG_ENTER();

    ...

    MuxOrch* mux_orch = gDirectory.get<MuxOrch*>();
    bool hw_config = isHwConfigured(neighborEntry);
   ...
}
```
The orch which are initialized here, if not directly needed, they are required by some other orch which are directly needed.

Comment (prsunny) [reply tests/mock_tests/flowcounterrouteorch_ut.cpp:183]: so do you need neighorch in this?

Comment (keboliu) [reply tests/mock_tests/flowcounterrouteorch_ut.cpp:183]: Yes, in order to successfully add a route entry, we need to add a neighbor entry beforehand.

Comment (prsunny) [reply tests/mock_tests/flowcounterrouteorch_ut.cpp:183]: Thanks for clarifying. I assume all orchs are relevant here.

**Review by prsunny — APPROVED** [review]

**Review by Junchao-Mellanox — APPROVED** [review]

**Pull request merged**